FEATURE(#246): ADD VOLUNTEER FORM

IsmaelP19 opened the pull request:
In this pr I created the volunteer form. 

### ⚠️ Important!
**Be sure to create a superuser first. You will have to create a NGO if none is created yet.**

The following features have been considered:

- The index page is not accessible yet (fixed on #324 )
- The create volunteer button on the volunteer list is not working yet (fixed on #302 )

You can play the next scenarios to check the form functionality:

- [x] You can navigate to the volunteer list. It should be empty now.
- [x] You can navigate to the volunteer creation form. To do so, please, go to http://localhost:8000/user/volunteer/create
- [x] If you haven't logged in, it will redirect you to the login page and after have logged in, the form will appear.
- [x] Check if colors are correct.
- [ ] Now, try the form on your own. Make as much tests as you want! 😉️

### PS: You should also be able to edit the volunteer created.

anac29 reviewed (commented):
I created a correct volunteer and no porblemas appeared. However, I tried to create one without email and it was succesfully created. Then I tried again to create a volunteer without email and an error ocurred, it said: that the email was duplicated the I assume the db is taking an empty string as an email. The only mandatory field is job so you can create volunteers with just that field completed. Nevertheless, this problem comes from backend, although it is always good to make some checks in frontend. So as a suggestion add a validator for the date field to ensure no one says is -5 years hahah. Good job!

IsmaelP19 commented:
@anac29 I made a custom script to validate the dates of the form. I was inspired on your script on #250 and made some changes due to the way I show the form. 
As of the email field, I spoke to backend and they don't even know if it will be a mandatory field, so I won't do anything with it until they decide. 
Now you are free to check any part of the form 😆️

anac29 reviewed (approved):
Things I  tried and resulted in errors:
- The add volunteer button doen't work.
- You can create a volunteer with a telephone number of less that 9 digits.
- The restriction says that the birth date must be before the start date of the contract, however you can introduce a date of birth after the current date.

IsmaelP19 commented:
The following this have been fixed: 
- The add volunteer button should work by now.
- The birth date must be before the current date.
- Add some margin on the y-axis, according to mockups (as said on #347 )
- Deleted cancel button, according to mockups (as said on #347 )
- Modified submit button, according to mockups (as said on #347 )

IsmaelP19 commented:
Since all changes have been taken into account, I'm merging this pr. Thanks for your thoroughly review @anac29 👏🏼
